topo/node/ceos/ceos.go

```diff
@@ -34,6 +37,88 @@ func (n *Node) Proto() *topopb.Node {
 	return n.pb
 }
 
+var (
+	spawner = defaultSpawner
+)
+
+func defaultSpawner(command string, timeout time.Duration, opts ...expect.Option) (expect.Expecter, <-chan error, error) {
+	return expect.Spawn(command, timeout, opts...)
+}
+
+func (n *Node) GenerateSelfSigned(ctx context.Context, ni node.Interface) error {
+	selfSigned := n.pb.GetConfig().GetCert().GetSelfSigned()
+	if selfSigned == nil {
+		log.Infof("%s - no cert config", n.pb.Name)
+		return nil
+	}
+	log.Infof("%s - generating self signed certs", n.pb.Name)
+	log.Infof("%s - Waiting for pod to be running", n.pb.Name)
+	w, err := ni.KubeClient().CoreV1().Pods(ni.Namespace()).Watch(ctx, metav1.ListOptions{
```

Comment:
Is there an equivalent to "kubectl wait --for=condition=Ready pods --all -A --timeout=1m" in the go API? Might be easier than using watch and receiving results over the channel.

Reply:
so this is how the command works internally